pkg/controller/egressgateway/egressgateway_controller.go

```diff
@@ -526,6 +534,17 @@ func fillDefaults(egw *operatorv1.EgressGateway, installation *operatorv1.Instal
 	}
 }
 
+// validateExternalNetwork validates if the specified external network exists.
+func validateExternalNetwork(ctx context.Context, cli client.Client, externalNetwork string) error {
+	instance := &crdv1.ExternalNetwork{}
+	key := types.NamespacedName{Name: externalNetwork}
+	err := cli.Get(ctx, key, instance)
```

Comment:
@song-jiang Do we need to assert that the external network exists? Is there an ordering requirement here?

If it's possible to create them in either order, we might want to allow that - it would make installation slightly easier.

Reply:
@caseydavenport I think no customer should create EGW before external networks. If we don't enforce it, a customer may create EGW with misspelled network names ["valid-name", "wrong-name"], EGW will still be created for "valid-name", but for "worng-name" it is not doing the right thing. This could lead to security issues.